core/promise/methods/noop/promise_processor_test.go

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (C) 2018 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package noop
+
+import (
+	"testing"
+
+	"time"
+
+	"github.com/mysteriumnetwork/node/core/promise"
+	"github.com/mysteriumnetwork/node/money"
+	"github.com/mysteriumnetwork/node/session"
+	"github.com/stretchr/testify/assert"
+)
+
+var _ session.PromiseProcessor = &PromiseProcessor{}
+
+func TestPromiseProcessor_Start_SendsBalanceMessages(t *testing.T) {
+	dialog := &fakeDialog{}
+	processor := &PromiseProcessor{
+		dialog:          dialog,
+		balanceInterval: time.Millisecond,
+	}
+
+	err := processor.Start(proposal)
+	assert.NoError(t, err)
+	waitForCondition(t, func() bool {
```

Comment:
fix these so they don't fail under -race

1) Either stop the processor and then assert for results
2) Do some indirect monitoring(maybe logs?)